classes/author/maps/Schema.inc.php

```diff
@@ -15,14 +15,29 @@
 
 use APP\author\Author;
 use Illuminate\Support\Enumerable;
+use PKP\security\Role;
 use PKP\services\PKPSchemaService;
+use PKP\core\PKPRequest;
 
 class Schema extends \PKP\core\maps\Schema
 {
     public Enumerable $collection;
 
     public string $schema = PKPSchemaService::SCHEMA_AUTHOR;
 
+    public array $userGroupLabels = [];
+
+    public function __construct(PKPRequest $request, \PKP\context\Context $context, PKPSchemaService $schemaService)
+    {
+        parent::__construct($request, $context, $schemaService);
+
+        $userGroupDao = \DAORegistry::getDAO('UserGroupDAO'); /** @var \UserGroupDAO $userGroupDao */
+        $authorUserGroups = $userGroupDao->getByRoleId($this->context->getId(), Role::ROLE_ID_AUTHOR)->toArray();
+        foreach ($authorUserGroups as $authorUserGroup) {
+            $this->userGroupLabels[$authorUserGroup->getId()] = $authorUserGroup->getLocalizedName();
+        }
```

Comment:
Instead of storing just the labels, keep the whole `$authorUserGroups` array as a property of the `Schema` map. That way, if we want to add more information to the API response in the future, we have the objects there.

Reply:
Fixed I think. Also removed `userGroupLabels`.